Projects/Server/Network/Packets/IncomingAccountPackets.cs

```diff
@@ -185,6 +188,150 @@ public static void CreateCharacter(NetState state, CircularBufferReader reader,
         }
     }
 
+    public static void KRCreateCharacter(NetState state, CircularBufferReader reader, int packetLength)
+    {
+        int flags = 0;
+
+        int length = packetLength;
+
+        int unk1 = reader.ReadInt32(); // Pattern
+        int charSlot = reader.ReadInt32();
+        var name = reader.ReadAscii(30);
+        var unknown1 = reader.ReadAscii(30); // "Unknow"
+
+        int profession = reader.ReadByte();
+        int clientFlags = reader.ReadByte();
+
+        int gender = reader.ReadByte();
+        int genderRace = reader.ReadByte();
+
+        var stats = new StatNameValue[]
+        {
+            new(StatType.Str, reader.ReadByte()),
+            new(StatType.Dex, reader.ReadByte()),
+            new(StatType.Int, reader.ReadByte())
+        };
+
+        int hue = reader.ReadInt16();
+        int unk5 = reader.ReadInt32(); // 0x00 0x00 0x00 0x00
+        int unk6 = reader.ReadInt32(); // 0x00 0x00 0x00 0x00	
+
+        var skills = new SkillNameValue[state.NewCharacterCreation ? 4 : 3];
+        skills[0] = new SkillNameValue((SkillName)reader.ReadByte(), reader.ReadByte());
+        skills[1] = new SkillNameValue((SkillName)reader.ReadByte(), reader.ReadByte());
+        skills[2] = new SkillNameValue((SkillName)reader.ReadByte(), reader.ReadByte());
+        if (state.NewCharacterCreation)
+        {
+            skills[3] = new SkillNameValue((SkillName)reader.ReadByte(), reader.ReadByte());
+        }
+
+        var unknown2 = reader.ReadAscii(25); // Pack of 0x00
+        int unk7 = reader.ReadByte(); // Another 0x00
+
+        int hairHue = reader.ReadInt16();
+        int hairID = reader.ReadInt16();
+
+        int unk8 = reader.ReadByte();
+        int unk9 = reader.ReadInt32();
+        int unk10 = reader.ReadByte();
+        int shirtHue = reader.ReadInt16();
+        int shirtID = reader.ReadInt16();
+        int unk13 = reader.ReadByte();
+        int faceColor = reader.ReadInt16();
+        int faceID = reader.ReadInt16();
+        int unk14 = reader.ReadByte();
+        int beardHue = reader.ReadInt16();
+        int beardID = reader.ReadInt16();
+
+        int cityIndex = 0; // Obsolete
+        int pantsHue = shirtHue; // Obsolete
+        Race race = null;
+        bool female = false;
+
+        female = (gender != 0);
+        race = Race.Races[(byte)(((genderRace - 1)))]; //SA client sends race packet one higher than KR, so this is neccesary
```

Comment:
Combine this with L248:
```cs
var race = Race.Races[genderRace - 1] ?? Race.DefaultRace;
```

And then delete the if statement for null race.

Reply:
Solved